### ISSUE TITLE
Found with fuzzer: relmon: mon not in list.

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#15]
```
rogram received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
#2  0x0000555555a4217c in NH_abort () at end.c:236
#3  0x0000555555a48f78 in panic (str=0x5555564d4fe0 "relmon: mon not in list.") at end.c:802
#4  0x0000555555da4ad6 in relmon (mon=0x6110017f6a00, 
    monst_list=0x5555567ba6e0 <migrating_mons>) at mon.c:3056
#5  0x000055555596c5de in migrate_to_level (mtmp=0x6110017f6a00, tolev=6 '\006', 
    xyloc=1 '\001', cc=0x0) at dog.c:867
#6  0x000055555596c34c in migrate_to_level (mtmp=0x6110017cdd80, tolev=6 '\006', 
    xyloc=1 '\001', cc=0x0) at dog.c:840
#7  0x0000555555ddaa62 in migrate_mon (mtmp=0x6110017cdd80, target_lev=6 '\006', 
    xyloc=1 '\001') at mon.c:4519
#8  0x0000555555ddaa12 in m_into_limbo (mtmp=0x6110017cdd80) at mon.c:4509
#9  0x00005555559688c2 in mon_arrive (mtmp=0x6110017cdd80, with_you=0 '\000') at dog.c:555
#10 0x0000555555967302 in losedogs () at dog.c:390
#11 0x000055555590e622 in goto_level (newlevel=0x7fffffffdf60, at_stairs=0 '\000', 
    falling=0 '\000', portal=0 '\000') at do.c:1996
#12 0x0000555555910a27 in deferred_goto () at do.c:2252
#13 0x000055555619718f in level_tele () at teleport.c:1380
#14 0x00005555558335ad in wiz_level_tele () at cmd.c:1102
#15 0x00005555558607db in rhack (cmd=0x55555679aa40 <in_line> "\026") at cmd.c:5515
#16 0x0000555555762c00 in moveloop (resuming=0 '\000') at allmain.c:798
#17 0x00005555563b3097 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737337632192, 93824997400142, 335544320, 
            140737341055120, 1073741824, 0, 140737342801568, 4294967295, 140737337360384, 
            140737342784672, 3331540481901294080}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342784672, 3331540481901294080, 
              140737488344112, 140737342801568, 18446744073709547520, 140737327112192, 
              140737345470680, 140737488345392, 17592186043162, 140737344339986, 
              17592186043014, 140737488338944, 140737341055120, 3331540481901294080, 
              140737488344512}}, sa_flags = -1030786560, sa_restorer = 0x7fffffffd780}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a4217c in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a48f78 in panic (str=0x5555564d4fe0 "relmon: mon not in list.") at end.c:802
        the_args = {{gp_offset = 8, fp_offset = 48, overflow_arg_area = 0x7fffffffd850, 
            reg_save_area = 0x7fffffffd780}}
#4  0x0000555555da4ad6 in relmon (mon=0x6110017f6a00, 
    monst_list=0x5555567ba6e0 <migrating_mons>) at mon.c:3056
        mtmp = 0x0
        mx = 0
        my = 0
        on_map = 0 '\000'
        unhide = 1 '\001'
#5  0x000055555596c5de in migrate_to_level (mtmp=0x6110017f6a00, tolev=6 '\006', 
    xyloc=1 '\001', cc=0x0) at dog.c:867
        obj = 0x0
        new_lev = {dnum = 104 'h', dlevel = -63 '\301'}
        xyflags = 0 '\000'
        num_segs = 0
#6  0x000055555596c34c in migrate_to_level (mtmp=0x6110017cdd80, tolev=6 '\006', 
    xyloc=1 '\001', cc=0x0) at dog.c:840
        obj = 0x555556727390 <rnglist+16>
        new_lev = {dnum = -32 '\340', dlevel = -39 '\331'}
        xyflags = 0 '\000'
        num_segs = 0
#7  0x0000555555ddaa62 in migrate_mon (mtmp=0x6110017cdd80, target_lev=6 '\006', 
    xyloc=1 '\001') at mon.c:4519
No locals.
#8  0x0000555555ddaa12 in m_into_limbo (mtmp=0x6110017cdd80) at mon.c:4509
        target_lev = 6 '\006'
        xyloc = 1 '\001'
#9  0x00005555559688c2 in mon_arrive (mtmp=0x6110017cdd80, with_you=0 '\000') at dog.c:555
        t = 0x0
        xlocale = 0 '\000'
        ylocale = 0 '\000'
        xyloc = 0 '\000'
        xyflags = 0 '\000'
        wander = 8 '\b'
        num_segs = 0
        failed_to_place = 1 '\001'
#10 0x0000555555967302 in losedogs () at dog.c:390
        mtmp = 0x6110017cdd80
        mtmp0 = 0x611001963dc0
        mtmp2 = 0x6110017f6a00
        dismissKops = 0
#11 0x000055555590e622 in goto_level (newlevel=0x7fffffffdf60, at_stairs=0 '\000', 
    falling=0 '\000', portal=0 '\000') at do.c:1996
        fd = 4
        l_idx = 0
        new_ledger = 6 '\006'
        cant_go_back = 0 '\000'
        great_effort = 0 '\000'
        up = 0 '\000'
        newdungeon = 0 '\000'
        was_in_W_tower = 0 '\000'
        familiar = 0 '\000'
        new = 0 '\000'
        mtmp = 0x0
        whynot = '\000' <repeats 255 times>
        annotation = 0x0
        dist = 2
        do_fall_dmg = 0 '\000'
#12 0x0000555555910a27 in deferred_goto () at do.c:2252
        dest = {dnum = 0 '\000', dlevel = 6 '\006'}
        typmask = 64
#13 0x000055555619718f in level_tele () at teleport.c:1380
        get_there_from = "get there from %s."
        mtmp = 0x0
        newlev = 6
        newlevel = {dnum = 0 '\000', dlevel = 6 '\006'}
        escape_by_flying = 0x0
        buf = '\000' <repeats 168 times>, "\340\330T\367\377\177\000\000\344\260;VUU", '\000' <repeats 73 times>
        force_dest = 0 '\000'
        ulev = 4
        vlev = 46
#14 0x00005555558335ad in wiz_level_tele () at cmd.c:1102
No locals.
#15 0x00005555558607db in rhack (cmd=0x55555679aa40 <in_line> "\026") at cmd.c:5515
        tlist = 0x55555665bcb0 <extcmdlist+5136>
        res = 27272704
        func = 0x55555583355e <wiz_level_tele>
        spkey = 0
        prefix_seen = 0 '\000'
        bad_command = 106 'j'
        firsttime = 1 '\001'
#16 0x0000555555762c00 in moveloop (resuming=0 '\000') at allmain.c:798
        moveamt = 10
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 34917
        past_clock = 68392
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#17 0x00005555563b3097 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
This bug is very likely to be caused by me adding the Lethe Branch as the second defined dungeon when the notes in dungeon.def clearly told me not to. I was able to integrate the lethe levels so they appear in the main dungeon after medusa.

**Comment by elunna** [comment]
fix for riding relmon?

the suggested fix was to place rider and steed together at the same time and not in separate calls

in src/dog.c:845 or so there's a bit that's commented with "Recursive call to levelport monster steeds"
the source of the problem

14:11 <qt> i was thinking the fix would be not to handle steeds just like "oh, these are also monsters on the exact same list", but to mark them so they only enter and leave the list simultaneous with their riders. or put them on a separate migrating_steeds list

14:11 <qt> but that sounds more complicated than what amateurhour was saying so idk if i am overthinking it

**Comment by elunna** [comment]
Possibly fixed by removing monster steeds in commit da87ef141.
